### PR TITLE
Fix Llama4 example

### PR DESCRIPTION
**Pull request by yiliu30**

### CODE REVIEW

File: torchao/prototype/moe_quant/llama4_quant.py [hunk 1]
```diff
@@ -58,7 +58,10 @@ def convert_fn(module):
 
 
 model_id = "meta-llama/Llama-4-Scout-17B-16E-Instruct"
-model = Llama4ForCausalLM.from_pretrained(model_id, torch_dtype=torch.bfloat16)
+
+dtype = torch.bfloat16
+torch.set_default_dtype(dtype)
```
Comment (yiliu30) [inline torchao/prototype/moe_quant/llama4_quant.py:63]: To make sure the input and model weights are the same dtype.

File: torchao/prototype/moe_quant/quantizable_moe_modules.py [hunk 2]
```diff
@@ -30,7 +30,7 @@ def __init__(
     def forward(self, x: Tensor) -> Tensor:
         batch_size = x.shape[0]
         x = x.view(-1, self.hidden_dim)  # x: [T, D]
-        scores = self.router(x)  # [T, E]
+        scores = self.router(x)[0]  # [T, E]
```
Comment (yiliu30) [inline torchao/prototype/moe_quant/quantizable_moe_modules.py:33]: The original router returns `router_scores` and `router_logits`.
https://github.com/huggingface/transformers/blob/19ffe0219dae122203f9726669f88ef1c6ea3bb4/src/transformers/models/llama4/modeling_llama4.py#L143

**Comment by yiliu30** [comment]
Hi @liangel-02 @andrewor14   Looks like the failed CI checks are not related to this PR. Could you help retrigger them?

File: torchao/prototype/moe_quant/quantizable_moe_modules.py (hunk 2 above)
Comment (jerryzh168) [reply torchao/prototype/moe_quant/quantizable_moe_modules.py:33]: router here is a nn.Linear, not `Llama4Router` actually, see L21?

Comment (yiliu30) [reply torchao/prototype/moe_quant/quantizable_moe_modules.py:33]: This `rounter` was replaced with the original rounter.
https://github.com/pytorch/ao/blob/9d01b43ecbaa14433762fd6787409808c022a8cd/torchao/prototype/moe_quant/llama4_quant.py#L48-L53

Comment (jerryzh168) [reply torchao/prototype/moe_quant/quantizable_moe_modules.py:33]: so this module does not run by itself? seems quite confusing

Comment (yiliu30) [reply torchao/prototype/moe_quant/quantizable_moe_modules.py:33]: The `MOEFeedForwardAOQuantizable` was used, but it seems that its `rounter` shouldn’t be quantized in order to preserve accuracy. Could you confirm that? @HDCharles

**Comment by jerryzh168** [comment]
@yiliu30 I'd recommend wait a bit until https://github.com/pytorch/ao/pull/2744 is landed for moe support

**Comment by yiliu30** [comment]
> @yiliu30 I'd recommend wait a bit until #2744 is landed for moe support

Sure, thanks for the information.
